tiledb/sm/query/query.cc

```diff
@@ -1475,6 +1489,12 @@ Status Query::check_buffer_names() {
 
 Status Query::create_strategy(bool skip_checks_serialization) {
   if (type_ == QueryType::WRITE || type_ == QueryType::MODIFY_EXCLUSIVE) {
+    // Set the fragment timestamp if it has not yet been set.
+    if (!current_fragment_timestamp_.has_value()) {
```

Comment:
We should either do this at init or here, not both.

Reply:
This needs to happen in `init()` for dimension labels to have the same timestamp in the array. The reason I added this here as well is because there is a few places `create_strategy` can be called before the Query is initialized.

1. In serialization the `Query::strategy` and `Query::reset_strategy_with_layout` can be used to create the strategy before initialization.
2. For remote arrays, `Query::submit` and `Query::submit_and_finalize` create the strategy but do not initialize the Query.

What I did is admittedly hacky. Do you think it would be better to explicitly set the timestamp in these 4 places? Or should I just document what I did and why better?